### PR TITLE
update death messages

### PR DESCRIPTION
**Pull request by Cahrs**

### CODE REVIEW

File: mods/news/init.lua [hunk 1]
```diff
@@ -1,46 +1,89 @@
...
         if reason.object then
-            killer = reason.object:get_player_name()
-            news = " BREAKING NEWS: Local man \"" .. obj:get_player_name() .. "\" " .. chat_reasons[reason.type][math.random(1, #chat_reasons[reason.type])] .. "\"" .. killer .. "\"."
+            if reason.object:is_player() then
+                killer = reason.object:get_player_name()
+            else
+                killer = reason.object:get_luaentity().name
+            end
+
+            news_msg = " BREAKING NEWS: Local man \"" .. player .. "\" " .. reason_msg .. " \"" .. killer .. "\"."
```
Comment (luk3yx) [inline mods/news/init.lua:76]:
```suggestion
            news_msg = " BREAKING NEWS: Local player \"" .. player .. "\" " .. reason_msg .. " \"" .. killer .. "\"."
```

File: mods/news/init.lua [hunk 2]
```diff
@@ -50,30 +93,39 @@ minetest.register_chatcommand("news", {
...
-        if station and news then
+        if station and news_msg then
             --print(station, news, bool)
                 
             station = minetest.colorize("#a8659c", "[" .. string.upper(station) .. " News" .. "]")
             
             if bool and bool == "$" then
-                news = string.sub(news, 1, #news-1)
-                news = "BREAKING NEWS: " .. news
+                news_msg = string.sub(news, 1, #news_msg-1)
```
Comment (luk3yx) [inline mods/news/init.lua:105]:
```suggestion
                news_msg = string.sub(news, 1, #news_msg - 1)
```

File: mods/news/init.lua [hunk 3]
```diff
@@ -50,30 +93,39 @@ minetest.register_chatcommand("news", {
...
 
-            minetest.chat_send_all(station .. " " .. news)
+            minetest.chat_send_all(station .. " " .. news_msg)
         else
             local station = "[BBC News] "
             station = minetest.colorize("#a8659c", station)
 
-            local news = "BREAKING NEWS: salt levels rise on PK."
-            news = minetest.colorize("#7f99b1", news)
+            local news_msg = "BREAKING NEWS: salt levels rise on PK."
```
Comment (luk3yx) [inline mods/news/init.lua:116]:
```suggestion
            local news_msg = "BREAKING NEWS: Salt levels rise on PK."
```

File: mods/news/init.lua [hunk 4]
```diff
@@ -50,30 +93,39 @@ minetest.register_chatcommand("news", {
...
+})
+
+news.register_deathmsg_tbl("nodes", "default:lava_source", 
+{
+    "melted into a ball of fire",
+    "couldn't resist the warm glow of lava",
+    "dug straight down"
+})
+
+news.register_deathmsg_tbl("nodes",
```
Comment (luk3yx) [inline mods/news/init.lua:131]: Did you forget to write more code or remove this?

**Review by luk3yx — CHANGES_REQUESTED** [review]
There are some formatting issues and missing code at the end of the file.

Is `news/init.lua` meant to be copied into the top-level directory as well?

**Comment by Cahrs** [comment]
Closing (reforked to fix some screwups)

**Pull request closed without merging**